apps/total-typescript/src/pages/books/[book]/index.tsx

```diff
@@ -66,8 +61,17 @@ const BookRoute: React.FC<{
   book: Book
   bookBody: MDXRemoteSerializeResult | null
 }> = ({book, bookBody}) => {
-  const {lastBookmarkedResource} = useBookProgress(book.slug.current)
-  const {bookmarks, refetch: refetchBookmarks} = useBookmarks(book.slug.current)
+  const {data: lastBookmarkedResource} =
+    trpc.bookmarks.lastBookmarkedResource.useQuery({
+      type: 'book',
+    })
+
+  const chaptersSlugs = book.chapters.map((chapter) => chapter.slug)
+  const {data: bookmarks} = trpc.bookmarks.getBookmarksForUser.useQuery()
+
+  const bookmarksForBook = bookmarks?.filter((bookmark) => {
+    return chaptersSlugs.includes(bookmark?.fields?.chapterSlug ?? '')
+  })
```

Comment:
get all of users bookmarks and then filter on the chapter slugs that are present in the current module. 

We can't do this check at the `resourceId` level because of how data is stored in sanity, `resourceId` for these bookmarks are `h2`s in the body of a chapter resource.

Reply:
imo from a data perspective we should be treating sections as "lessons" and have them stand as Content Resources and not formatted text in a document so they can be chunked/referenced/indexed at that level